File: java_program/src/SimpleProgram.java
import java.io.IOException;
import java.lang.instrument.Instrumentation;
import java.util.ArrayList;
import java.util.Date;
import java.util.Random;
import java.util.Scanner;


public class SimpleProgram {
    static int INSTANCE_NUMBER = 250000;
    static int RUN_NUMBER = 20;
    static String[] BRANDS = { "BMW", "Audi", "VW", "Chevrolet", "Ford" };


    public static void runProgram() throws InterruptedException {
        ArrayList<Car> carPool = new ArrayList<>();
        //Scanner input = new Scanner(System.in);
        //input.next();
        //System.out.println("Starting in 20 sec");
        Thread.sleep(15000);
        long start_time = System.nanoTime();


        for (int k = 0; k < RUN_NUMBER; k++) {

            for(int i = 0; i < INSTANCE_NUMBER; i++) {

                Random generator = new Random();
                int hp = generator.nextInt(1000) + 100;
                int weight = generator.nextInt(3000) + 1000;
                int brand_index = generator.nextInt(BRANDS.length);
                String brand = BRANDS[brand_index];

                Car new_car = new Car(weight, hp, 4, brand, 100000, 5, false);
                int time = generator.nextInt(20) + 1;
                new_car.accelerate(time);
                carPool.add(new_car);

            }

            for(Car car: carPool) {
                car.accelerate(5);
                //System.out.print("\n\nVertical normal force on 90 degree angle: " + car.getVerticalNormalForce());
                // System.out.print("\nSpeed: " + car.getSpeed() + " km/h");
            }
        }

        long end_time = System.nanoTime();
        long time_elapsed = end_time - start_time;
        System.out.print("Time in seconds: ");
        System.out.print(time_elapsed/1000000000);

        //System.out.println("Press Enter to stop...");
        //System.in.read();
    }



    public static void main(String[] args) throws InterruptedException, IOException {

        runProgram();
    }
}

File: python_program/simpleProgram.py
__author__ = 'KasperKBerg'


import random
import time
import cProfile
import datetime
import stackimpact
from memory_profiler import profile
#import psutil
import sys

class Vehicle:
    __weight_kg= None
    __horsepowers= None
    __number_wheels= None
    __brand= None
    __price_usd = None
    gravity_constant = 9.81

    def __init__(self, weight_kg, horsepowers, number_wheels, brand, price_usd):
        self.__weight_kg = weight_kg
        self.__horsepowers = horsepowers
        self.__number_wheels = number_wheels
        self.__brand = brand
        self.__price_usd = price_usd


    def getVerticalNormalForce(self):
        return self.__weight_kg * self.gravity_constant


    def getVerticleNormalForceOfVehicle(self, vehicle):
        return self.__weight_kg * self.gravity_constant


    def equals(self,object):

        if(object == None):
            return False

        if(not(True)): ##object instanceof Vehicle
            return False
        return False

class Car(Vehicle):
    __acceleration_factor= None
    is_veteran= None
    __speed = None


    def __init__(self,weight_kg, horsepowers, number_wheels, brand, price_usd, acceleration_factor, is_veteran):
        super().__init__(weight_kg, horsepowers, number_wheels, brand, price_usd)
        self.__acceleration_factor = acceleration_factor
        self.is_veteran = is_veteran
        self.__speed = 0.0

    def accelerate(self,sec):
        self.__speed+= self.__acceleration_factor*sec

    def declerate(self,sec):
        if (self.__speed>0):
            self.__speed-=self.__acceleration_factor*sec
    def getSpeed(self):
        return self.__speed



INSTANCE_NUMBER = 250000
RUN_NUMBER = 20
BRANDS = ["BMW", "Audi", "VW", "Chevrolet", "Ford"]


def createCars(carPool):
    for i in range(RUN_NUMBER):
        for i in range(0, INSTANCE_NUMBER):
            generator = random
            hp = generator.randint(100, 1100)
            weight = generator.randint(100, 1100)
            brand_index = generator.randint(0, len(BRANDS) - 1)
            brand = BRANDS[brand_index]
            new_car = Car(weight, hp, 4, brand, 100000, 5, False)
            time = generator.randint(1, 21)
            new_car.accelerate(time)
            carPool.append(new_car)
        for car in carPool:
            car.accelerate(5)
         #   print("\n\nVertical normal force on 90 degree angle: " + str(car.getVerticalNormalForce()))
          #  print("\nSpeed: " + str(car.getSpeed()) + " km/h")


## @profile  will enable memory profiler
#@profile
def run_program():
    carPool = []
    createCars(carPool)
    print('\n')
    # span.stop()

def intSizetest():
    x = 2
    print(sys.getsizeof(2))



if __name__ == '__main__':

    start_time = datetime.datetime.now()
    run_program()
    print('\nFINISHED AT: ')
    print(datetime.datetime.now() - start_time)
    print('\n')



